controllers/drplacementcontrol_controller.go

```diff
@@ -888,31 +888,47 @@ func (r *DRPlacementControlReconciler) getVRGsFromManagedClusters(drpc *rmn.DRPl
 	annotations[DRPCNameAnnotation] = drpc.Name
 	annotations[DRPCNamespaceAnnotation] = drpc.Namespace
 
+	var failedClusterToQuery string
+
+	var clustersQueriedSuccessfully int
+
 	for _, drCluster := range rmnutil.DrpolicyClusterNames(drPolicy) {
 		vrg, err := r.MCVGetter.GetVRGFromManagedCluster(drpc.Name, drpc.Namespace, drCluster, annotations)
 		if err != nil {
 			// Only NotFound error is accepted
 			if errors.IsNotFound(err) {
 				log.Info(fmt.Sprintf("VRG not found on %q", drCluster))
+				clustersQueriedSuccessfully++
 
 				continue
 			}
 
-			if drpc.Spec.Action == rmn.ActionFailover && drpc.Spec.FailoverCluster != drCluster {
-				log.Info(fmt.Sprintf("Skipping fetching VRG from %s due to failure. Error (%v)",
-					drCluster, err))
-
-				continue
-			}
+			failedClusterToQuery = drCluster
 
-			return vrgs, fmt.Errorf("failed to retrieve VRG from %s. err (%w)", drCluster, err)
+			log.Info(fmt.Sprintf("failed to retrieve VRG from %s. err (%v)", drCluster, err))
 		}
 
+		clustersQueriedSuccessfully++
+
 		vrgs[drCluster] = vrg
 
 		log.Info("VRG location", "VRG on", drCluster)
 	}
 
+	if clustersQueriedSuccessfully == 0 {
+		return vrgs, fmt.Errorf("failed to retrieve VRGs from clusters")
+	}
+
+	if len(vrgs) == 0 && failedClusterToQuery != drpc.Spec.FailoverCluster {
```

Comment:
If we failover to C2 and assume DRPC is:
```
	failoverCluster: C2
        preferredCluster: C1
```
- If now C1 is down and hub as well
- We will find failedClusterToQuery != failoverCluster to be false and not enter the fixup?

Reply:
If hub is down and we recover it somewhere else, we will enter the fixup phase regardless.

Reply:
Sorry, changing previous comment to say:
`If now C2 is down and hub as well` instead of C1